app/src/org/commcare/models/FormRecordProcessor.java

```diff
@@ -85,7 +85,8 @@ public TransactionParser getParser(KXmlParser parser) {
 
         //Let anyone who is listening know!
         Intent i = new Intent("org.commcare.dalvik.api.action.data.update");
-        this.c.sendBroadcast(i);
+        i.putStringArrayListExtra("cases", factory.getCreatedAndUpdatedCases());
```

Comment:
Interesting... We do treat case Id's as private data generally. IS it clear whether this broadcast is permissioned? There's an Android "read CommCare data" permission that we require for the content provider. If it's set for the broadcasts as well this seems good to go.